src/main/java/frc/robot/subsystems/colour_wheel/commands/FindColour.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.subsystems.colour_wheel.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.colour_wheel.ColourWheel;
+
+/**
+ * Spin the wheel until you reach the desired color.
+ */
+public class FindColour extends CommandBase {
+    private ColourWheel colourWheel;
+    private String colour;
+    private double power;
+    private String[] colours = new String[]{"YELLOW", "BLUE", "GREEN", "RED"};
+    private String tempColour;
+    private int resultColorDis;//Amount of colors away from initial color to target color.
+    private int previousColor;
+    private boolean isNewColorSeen = false;
+
+    public FindColour(ColourWheel colourWheel, String colour, double power) {
+        this.colourWheel = colourWheel;
+        this.colour = colour;
+        this.power = power;
+        addRequirements(colourWheel);
+    }
+
+    @Override
+    public void initialize() {

```

Comment:
There is way too much code in this function, and it's not clear what the code does. Split the code into a few more methods and rename the variables.